android/uhabits-android/src/main/java/org/isoron/uhabits/intents/IntentParser.kt

```diff
@@ -36,14 +36,19 @@ class IntentParser
         return CheckmarkIntentData(parseHabit(uri), parseTimestamp(intent))
     }
 
+    fun copyIntentData(source: Intent, destination: Intent){
+        destination.data=source.data;
+        destination.putExtra("timestamp",source.getLongExtra("timestamp",getToday()))
```

Comment:
Please make sure your code is formatted consistently. We use spaces before `{`, after `,` and around `=`.